RecoTauTag/RecoTau/plugins/RecoTauPiZeroTrivialPlugin.cc

```diff
@@ -41,8 +41,8 @@ namespace reco::tau {
     output.reserve(pfGammaCands.size());
 
     for (auto const& gamma : pfGammaCands) {
-      std::unique_ptr<RecoTauPiZero> piZero(
-          new RecoTauPiZero(0, (*gamma).p4(), (*gamma).vertex(), 22, 1000, true, RecoTauPiZero::kTrivial));
+      auto piZero =
```

Comment:
why all these by pointer and not by value?

Reply:
Hmm, I am not an original author of this code (and other pointed examples) so I can only guess that it was to move instead of copying of those objects (which actually does not require pointers, right?). It can be redone, but I would like to avoid it if there is not any important reason, as it requires  some effort because it will require changes in other places.

Reply:
The object can be built in place. You are indeed right that it would require (extensive?) changes in all clients...

Reply:
OK. It there any performance reason to change it? Extensiveness of changes is partly matter of taste (at least a few other modules have to be touched), but I prefer to not change it without really good reason.

Reply:
one level of indirection  and maybe less memory fragmentation. It also depends how it is used and if the class can be finalized. Core ( @makortel ) may help in the decision.

Reply:
The vector is returned by value: not sure that the copy is elided...

Reply:
Given the extensiveness I'd maybe make a new PR, as the changes of this PR are useful as such.

Reply:
OK, so I will try to fix it as part of other PR. However, in this one I will try to fix similar, but less extensive, issue in `RecoTauTag/RecoTau/plugins/TauDiscriminantCutMultiplexer.cc` mentioned below.